### PR TITLE
[ENH] Validate bids

### PR DESCRIPTION
**Pull request by arnaudbore**
With this PR tractoflow (some work need to be done) will be able to handle new multiband acquisitions as well as full AP/PA acquisitions.

Here are some data to test (braindata): `/braindata/processedData/data_for_validate_bids`
DO NOT SHARE THESE DATA. Thank you

New structure output:

>     {
>         "DWIPhaseEncodingDir": "",
>         "TotalReadoutTime": 0.062,
>         "bval": "",
>         "bvec": "",
>         "dwi": "",
>         "rev_DWIPhaseEncodingDir": "",
>         "rev_bval": "",
>         "rev_bvec": "",
>         "rev_dwi": "",
>         "rev_topup": "",
>         "run": ,
>         "session": "",
>         "subject": "",
>         "t1": "",
>         "topup": ""
>     },
>

### CODE REVIEW

**Comment by scil-admin** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-591/1/

**Comment by scil-admin** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-591/2/

**Comment by scil-admin** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-591/3/

**Comment by scil-admin** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-591/4/

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Review by EmmaRenauld — COMMENTED** [review]
- No file dataset_description: the error is there. `ValueError: 'dataset_description.json' is missing from project root. Every valid BIDS dataset must have this file.` I tried with an empty 'dataset_description.json' file. The error was very unclear (`json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
`). But as soon as there was at least the { }, the errors were better (ex:  `Mandatory 'Name' field missing from dataset_description.json.`).
- I tried in a folder with only a derivatives files: output is `[]`
- No success here in having my subject discovered... The script runs but resulting json is always `[]`. Tried with or without participants.tsv, tried with name sub-001 or a random name. Tried in a sourcedata folder or not. Not sure what to change to make it discovered.

So in summary I have:
root
    |- derivatives
    |- dataset_description.json
    |- participants.tsv
    |- my_subject  -> t1, dwi, bval, bvec

P.S. The help of this script does not contain enough `\n` compared to what we usually do + says  `you want to be be analyzed`.

P.s.2 Would we like to link this page? https://scil-documentation.readthedocs.io/en/latest/heavy/storage.html

File: scripts/scil_validate_bids.py [hunk 1]
```diff
@@ -56,12 +56,13 @@ def get_metadata(bf):
     Dictionnary containing the metadata
     """
     filename = bf.path.replace(
```
[the remark this answers is not in the document]
Comment (arnaudbore) [reply scripts/scil_validate_bids.py:58]: Quick answer no, there are a couple of metadata that are mandatory and specific to each dwi sessions. I'm thinking about "IntendedFor" for example.

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

File: scripts/scil_validate_bids.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexVCaron) [reply scripts/scil_validate_bids.py:58]: I see. All good for me.

**Review by AlexVCaron — APPROVED** [review]
Everything is good for me. Some sections could be cleaner, but this can be let for future work.

File: scripts/scil_validate_bids.py [hunk 2]
```diff
@@ -143,83 +158,122 @@ def get_data(nSub, dwi, t1s, associations, default_readout, clean):
...
-    dwi_metadata = get_metadata(dwi)
-    if 'PhaseEncodingDirection' in dwi_metadata:
-        dwi_PE = dwi_metadata['PhaseEncodingDirection']
-        dwi_PE = dwi_PE.replace(dwi_PE[0], conversion[dwi_PE[0]])
-        if len(dwi_PE) == 1:
-            dwi_revPE = dwi_PE + '-'
+    if 'run' in dwi[0].entities:
+        nRun = dwi[0].entities['run']
+
+    for index, curr_dwi in enumerate(dwi):
```
Comment (AlexVCaron) [inline scripts/scil_validate_bids.py:177]: There should be a validation that there is only 2 items in dwi.

File: scripts/scil_validate_bids.py [hunk 3]
```diff
@@ -228,31 +282,91 @@ def get_data(nSub, dwi, t1s, associations, default_readout, clean):
...
+        Current subject to analyse
+    Returns
+    -------
+    all_dwis: list
+        List of dwi
+    """
+    all_dwis = []
+    if layout.get_sessions(subject=nSub):
+        for curr_sess in layout.get_sessions(subject=nSub):
+            dwis = layout.get(subject=nSub,
```
[the remark this answers is not in the document]
Comment (arnaudbore) [reply scripts/scil_validate_bids.py:314]: The main diffrence is the use of a session. Extracting it into a function won't make is shorter.

[the remark this answers is not in the document]
Comment (AlexVCaron) [reply scripts/scil_validate_bids.py:314]: You could use something like an `extract_dwi_runs(subject, ..., **kwargs)` (can be a local function to associate_dwi) and pass session through kwargs if there is some. It is called in the for loop and in the else block. I would prefer it, but you decide.

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-591/11/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Review by AlexVCaron — APPROVED** [review]
Retested and everything works with my data
